docs/tutorials/customize_dataset.md

```diff
@@ -49,36 +49,107 @@ There are three necessary keys in the json file:
 
 In `configs/my_custom_config.py`:
 
 ```python
-...
-# dataset settings
+
+# the new config inherits the base configs to highlight the necessary modification
+_base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
+
+# 1. dataset settings
 dataset_type = 'CocoDataset'
 classes = ('a', 'b', 'c', 'd', 'e')
-...
 data = dict(
     samples_per_gpu=2,
     workers_per_gpu=2,
     train=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/train/data',
-        ...),
+        ann_file='path/to/your/train/annotation_data',
+        img_prefix='path/to/your/train/image_data'),
     val=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/val/data',
-        ...),
+        ann_file='path/to/your/val/annotation_data',
+        img_prefix='path/to/your/val/image_data'),
     test=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/test/data',
-        ...))
-...
+        ann_file='path/to/your/test/annotation_data',
+        img_prefix='path/to/your/test/image_data'))
+
+# 2. model settings
+
+# explicitly over-write all the `num_classes` field from default 80 to 5.
+model = dict(
+    roi_head=dict(
+        bbox_head=[
+            dict(
+                type='Shared2FCBBoxHead',
+                # explicitly over-write all the `num_classes` field from default 80 to 5.
+                num_classes=5),
+            dict(
+                type='Shared2FCBBoxHead',
+                # explicitly over-write all the `num_classes` field from default 80 to 5.
+                num_classes=5),
+            dict(
+                type='Shared2FCBBoxHead',
+                # explicitly over-write all the `num_classes` field from default 80 to 5.
+                num_classes=5)],
+    # explicitly over-write all the `num_classes` field from default 80 to 5.
+    mask_head=dict(num_classes=5)))
+```
+
+#### 2. Check the annotations of the customized dataset
+
+Assuming your customized dataset is COCO format, make sure you have the correct annotations in the customized dataset:
+
+1. The length for `categories` field in annotations should exactly equal the tuple length of `classes` fields in your config, meaning the number of classes (e.g. 5 in this example).
+2. The `classes` fields in your config file should have exactly the same elements with the `name` in `categories` of annotations. MMDetection automatically maps the uncontinuous `id` in `categories` to the continuous label indices. So it is the string order of `name` in `categories` field that matters, while the string order of `classes` in config does not matters.
```

Comment:
The string order of `classes` in config matters because during visualization it may use the class order in config. Check this.

Reply:
Yeah, the string order of `classes` in config matters because it is used during the visualization of predicted bounding boxes.
So the `name` in `categories` field should have exactly the same order with `classes` field in your config file.